### PR TITLE
PWGHF: Implementation of mult definition T0M and adding track selection

### PR DESCRIPTION
**Pull request by Simple-Shyam**
I modified the definition for the calculation of multiplicity from V0M to T0M and also added the track selection in the task.

### CODE REVIEW

**Review by fcolamar — COMMENTED** [review]
The updates are fine to me (my comments were posted on the parallel PR, just closed, and implemented there).
For the future, take into consideration what we just discussed internally on the most suited way of preparing commits and PR

**Comment by Simple-Shyam** [comment]
> ake into consideration what we just discussed internally on the most suited way of prepa

Ciao Fabio, Thank you for help in solving the issue related to merge request. I will follow it in my future commits.

**Review by fcolamar — APPROVED** [review]

File: PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx [hunk 1]
```diff
@@ -77,7 +77,7 @@ struct HfCorrelatorDplusHadronsDplusSelection {
   Partition<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi, aod::HfCand3ProngMcRec>> selectedDplusCandidatesMc = aod::hf_sel_candidate_dplus::isSelDplusToPiKPi >= selectionFlagDplus;
 
   void processDplusSelectionData(aod::Collision const& collision,
-                                 soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi> const&)
+                                 soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi> const& candidates)
```
Comment (vkucera) [inline PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:80]: Do not revert fixes.

File: PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx [hunk 2]
```diff
@@ -99,7 +99,7 @@ struct HfCorrelatorDplusHadronsDplusSelection {
   PROCESS_SWITCH(HfCorrelatorDplusHadronsDplusSelection, processDplusSelectionData, "Process Dplus Selection Data", false);
 
   void processDplusSelectionMcRec(aod::Collision const& collision,
-                                  soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi, aod::HfCand3ProngMcRec> const&)
+                                  soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi, aod::HfCand3ProngMcRec> const& candidates)
```
Comment (vkucera) [inline PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:102]: Do not revert fixes.

File: PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx [hunk 3]
```diff
@@ -123,7 +123,7 @@ struct HfCorrelatorDplusHadronsDplusSelection {
   }
   PROCESS_SWITCH(HfCorrelatorDplusHadronsDplusSelection, processDplusSelectionMcRec, "Process Dplus Selection MCRec", false);
 
-  void processDplusSelectionMcGen(aod::McCollision const&,
+  void processDplusSelectionMcGen(aod::McCollision const& mcCollision,
```
Comment (vkucera) [inline PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:126]: Do not revert fixes.

File: PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx [hunk 4]
```diff
@@ -239,15 +239,15 @@ struct HfCorrelatorDplusHadrons {
   }
 
   /// Dplus-hadron correlation pair builder - for real data and data-like analysis (i.e. reco-level w/o matching request via MC truth)
-  void processData(soa::Join<aod::Collisions, aod::Mults>::iterator const& collision,
-                   aod::TracksWDca const& tracks,
-                   soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi> const&, aod::BCsWithTimestamps const&)
+  void processData(SelCollisionsWithDplus::iterator const& collision,
+                   TracksWithDca const& tracks,
+                   CandidatesDplusData const& candidates, aod::BCsWithTimestamps const&)
```
Comment (vkucera) [inline PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:244]: Do not revert fixes.

File: PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx [hunk 5]
```diff
@@ -438,7 +444,7 @@ struct HfCorrelatorDplusHadrons {
     int counterDplusHadron = 0;
     registry.fill(HIST("hMCEvtCount"), 0);
 
-    auto getTracksSize = [&mcParticles](aod::McCollision const& /*collision*/) {
+    auto getTracksSize = [&mcParticles](aod::McCollision const& collision) {
```
Comment (vkucera) [inline PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:447]: Do not revert fixes.

File: PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx (hunk 1 above)
Comment (Simple-Shyam) [reply PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:80]: What should I do here? I didn't change anything here.

Comment (vkucera) [reply PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:80]: Fixed again in https://github.com/AliceO2Group/O2Physics/pull/5757

Comment (vkucera) [reply PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:80]: Correct

**Pull request merged**